xc7/utils/prjxray_arch_import.py

```diff
@@ -176,6 +179,149 @@ def create_synth_constant_tiles(
     })
 
 
+def get_phy_tiles(conn, tile_pkey):
+    """ Returns the locations of all physical tiles for specified tile. """
+    c = conn.cursor()
+    c2 = conn.cursor()
+
+    phy_locs = []
+    for (phy_tile_pkey, ) in c.execute(
+            "SELECT phy_tile_pkey FROM tile_map WHERE tile_pkey = ?",
+        (tile_pkey, )):
+        c2.execute(
+            "SELECT grid_x, grid_y FROM phy_tile WHERE pkey = ?",
+            (phy_tile_pkey, )
+        )
+        loc = c2.fetchone()
+        phy_locs.append(grid_types.GridLoc(*loc))
+
+    return phy_locs
+
+
+def is_in_roi(conn, roi, tile_pkey):
+    """ Returns if the specified tile is in the ROI. """
+    phy_locs = get_phy_tiles(conn, tile_pkey)
+    return any(roi.tile_in_roi(loc) for loc in phy_locs)
+
+
+def get_tile_prefix(conn, g, tile_pkey, site_as_tile_pkey):
+    """ Returns tile prefix of specified tile. """
+    phy_locs = get_phy_tiles(conn, tile_pkey)
+
+    prefix_tile = None
+
+    # If this tile has multiples phy_tile's, make sure only one has bitstream
+    # data, otherwise the tile split was invalid.
+    for loc in phy_locs:
```

Comment:
There should never be multiple physical locations for a VPR tile. After the split a SLICE may correspond to only one physical CLB.

Reply:
If we merge two tiles, then it can be true.  For example, INT_L merged on CLBLL_L.